diff --git a/setuptools/__init__.py b/setuptools/__init__.py
index 83882511..6e762e3d 100644
--- a/setuptools/__init__.py
+++ b/setuptools/__init__.py
@@ -251,3 +251,8 @@ class sic(str):
 
 # Apply monkey patches
 monkey.patch_all()
+def setup(**attrs):
+    # Make sure we have any requirements needed to interpret 'attrs'.
+    if not os.environ.get("dump_setup_attrs", None):
+        _install_setup_requires(attrs)
+    return distutils.core.setup(**attrs)
